### PR TITLE
implement order hint design

### PR DESCRIPTION
**Pull request by ramsayleung**
as title

### CODE REVIEW

File: src/main/java/io/reactivesw/category/application/controller/CategoryController.java [hunk 1]
```diff
@@ -27,6 +26,7 @@
 /**
  * Created by Davis on 16/11/18.
```
Comment (kaka89) [inline src/main/java/io/reactivesw/category/application/controller/CategoryController.java]: delete the author.

File: src/main/java/io/reactivesw/category/application/model/action/SetOrderHint.java [hunk 2]
```diff
@@ -13,12 +13,17 @@
...
-   * get update service name.
+   * The next order hint
+   */
+  private String nextOrderHint;
+
+  /**
+   * Get update service name.
    *
    * @return update service name
    */
```
Comment (davis999) [inline src/main/java/io/reactivesw/category/application/model/action/SetOrderHint.java]: delete author comments.

File: src/main/java/io/reactivesw/category/domain/service/update/SetOrderHintService.java [hunk 3]
```diff
@@ -8,19 +8,31 @@
...
+  /**
+   * Calculate median of two order hint
+   * @param previousOrderHint previousOrderHint
+   * @param nextOrderHint nextOrderHint
+   * @return median of two order hint
+   */
+  private String calculateMedianOfOrderHint(String previousOrderHint, String nextOrderHint) {
+    double previous = Double.parseDouble(previousOrderHint);
+    double next = Double.parseDouble(nextOrderHint);
+    return String.valueOf((previous + next) / 2);
```
Comment (davis999) [inline src/main/java/io/reactivesw/category/domain/service/update/SetOrderHintService.java]: use BigDecimal!!!

File: src/main/java/io/reactivesw/category/domain/service/CategoryService.java [hunk 4]
```diff
@@ -285,4 +274,16 @@ private Category getParentCategory(String parentId) {
...
+   * @return order hint
+   */
+  private String getOrderHint() {
+    long currentTime = System.currentTimeMillis();
+    int length = String.valueOf(currentTime).length();
+    double divisor = Math.pow(10, length);
+//    convert current time to decimal
+    return String.valueOf(currentTime / divisor);
+  }
 }
```
Comment (davis999) [inline src/main/java/io/reactivesw/category/domain/service/CategoryService.java]: move to another util class.

Comment (davis999) [reply src/main/java/io/reactivesw/category/domain/service/CategoryService.java]: I think "0." + currentTime is more simple.

File: src/main/java/io/reactivesw/category/application/controller/CategoryController.java [hunk 5]
```diff
@@ -111,7 +107,6 @@ public CategoryView getCategoryById(@PathVariable(value = Router.CATEGORY_ID) St
 
   /**
    * Query category list.
-   *
    * @return the list
    */
   @GetMapping(Router.CATEGORY_ROOT)
```
Comment (davis999) [inline src/main/java/io/reactivesw/category/application/controller/CategoryController.java]: why change comments?

File: src/main/java/io/reactivesw/category/domain/service/CategoryService.java (hunk 4 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/category/domain/service/CategoryService.java]: In my opinion, it will more robust than simply use "0."+currentTime

**Comment by ramsayleung** [comment]
Delete author and date in comment, move `calculateMedianOfOrderHint` function to `categoryUtils` class, and refactor `calculateMedianOfOrderHint` function with `bigDecimal`

**Review by kaka89 — COMMENTED** [review]
change your version to 0.0.4-dev as dev branch

**Pull request closed without merging**